src/main/java/mekanism/common/item/gear/ItemMekaTool.java

```diff
@@ -235,18 +239,30 @@ public boolean onBlockStartBreak(ItemStack stack, BlockPos pos, Player player) {
             boolean silk = isModuleEnabled(stack, MekanismModules.SILK_TOUCH_UNIT);
             FloatingLong energyRequired = getDestroyEnergy(stack, state.getDestroySpeed(world, pos), silk);
             if (energyContainer.extract(energyRequired, Action.SIMULATE, AutomationType.MANUAL).greaterOrEqual(energyRequired)) {
+                Set<BlockPos> blocks = null;
+
+                //Setup initial set for blasting
+                IModule<ModuleBlastingUnit> blastingUnit = getModule(stack, MekanismModules.BLASTING_UNIT);
+                if (blastingUnit != null && blastingUnit.isEnabled() && !(state.getBlock() instanceof BlockBounding) && !player.isShiftKeyDown()) {
+                    Direction direction = getPlayerPOVHitResult(world, player, ClipContext.Fluid.NONE).getDirection();
+                    blocks = ModuleBlastingUnit.findPositions(world, pos, direction, blastingUnit.getCustomInstance().getBlastRadius());
+                }
+
                 IModule<ModuleVeinMiningUnit> veinMiningUnit = getModule(stack, MekanismModules.VEIN_MINING_UNIT);
                 //Even though we now handle breaking bounding blocks properly, don't allow vein mining them
                 if (veinMiningUnit != null && veinMiningUnit.isEnabled() && !(state.getBlock() instanceof BlockBounding)) {
-                    boolean isOre = state.is(MekanismTags.Blocks.ATOMIC_DISASSEMBLER_ORE);
-                    //If it is extended or should be treated as an ore
-                    if (isOre || veinMiningUnit.getCustomInstance().isExtended()) {
-                        //Don't include bonus energy required by efficiency modules when calculating energy of vein mining targets
-                        FloatingLong baseDestroyEnergy = getDestroyEnergy(silk);
-                        Set<BlockPos> found = ModuleVeinMiningUnit.findPositions(state, pos, world, isOre ? -1 : veinMiningUnit.getCustomInstance().getExcavationRange());
-                        MekanismUtils.veinMineArea(energyContainer, world, pos, (ServerPlayer) player, stack, this, found,
-                              hardness -> getDestroyEnergy(baseDestroyEnergy, hardness), distance -> 0.5 * Math.pow(distance, isOre ? 1.5 : 2), state);
+                    if (blocks == null) {
+                        blocks = new HashSet<>();
+                        blocks.add(pos.immutable());
```

Comment:
I believe this could just be `blocks = Set.of(pos.immutable());` given we set `blocks` to the result of `findPositions` and we don't modify the passed in set inside of `findPositions`.